fedora_messaging/_session.py

```diff
@@ -180,13 +180,18 @@ def _connect_and_publish(self, exchange, message):
             if self._confirms:
                 self._channel.confirm_delivery()
 
+        routing_key = message._encoded_routing_key
+        if config.conf["topic_prefix"]:
+            routing_key = (
+                config.conf["topic_prefix"].encode("utf-8") + b"." + routing_key
```

Comment:
Wherever you choose to make the real routing key, it'll probably be a nicer user experience to make sure the topic prefix doesn't include a trailing ``.``. Something like:

 ``".".join((config.conf["topic_prefix"].rstrip("."), routing_key))``

Reply:
Indeed, that would avoid accidents.